packages/devextreme/testing/tests/DevExpress.ui.widgets.form/form.tests.js

```diff
@@ -4540,6 +4541,98 @@ QUnit.test('form should be dirty when some editors are dirty', function(assert)
     assert.strictEqual(form.option('isDirty'), false, 'form is not dirty when all editors are back to pristine');
 });
 
+[true, false].forEach((openOnFieldClick) => {
+    [true, false].forEach((hideOnOutsideClickValue) => {
+        QUnit.test(`Opened DropDownList must hide on input label click, openOnFieldClick: ${openOnFieldClick}, hideOnOutsideClick: ${hideOnOutsideClickValue} (T1257945)`, function(assert) {
+            let initialHideOnOutsideClick;
+            const $form = $('#form').dxForm({
+                formData: { CustomerID: 'VINET' },
+                items: [{
+                    itemType: 'group',
+                    colCount: 2,
+                    items: [{
+                        dataField: 'CustomerID',
+                        editorType: 'dxSelectBox',
+                        editorOptions: {
+                            items: ['VINET', 'VALUE', 'VINS'],
+                            value: '',
+                            openOnFieldClick,
+                            dropDownOptions: {
+                                hideOnOutsideClick: () => hideOnOutsideClickValue,
```

Comment:
[refactoring]
I suggest u renaming the parameter and pass a pure boolean here (and functions if u think it's important to be tested)
```suggestion
                                hideOnOutsideClick
```

Reply:
in that case initialHideOnOutsideClick we spy on is not a function and test fails on spy